tests/ProductsIndexingTest.php

```diff
@@ -91,4 +78,24 @@ public function testDefaultIndexableAttributes()
         $extraAttributes = implode(', ', array_keys($hit));
         $this->assertTrue(empty($hit), 'Extra products attributes ('.$extraAttributes.') are indexed and should not be.');
     }
+
+    public function testIfIndexingCanBeEnabledAndDisabled()
+    {
+        setConfig('algoliasearch/credentials/enable_backend', '0');
+
+        $productIndexer = new Algolia_Algoliasearch_Model_Indexer_Algolia();
+        $this->processTest($productIndexer, 'products', 0);
+
+        setConfig('algoliasearch/credentials/enable_backend', '1');
+
+        $productIndexer = new Algolia_Algoliasearch_Model_Indexer_Algolia();
+        $this->processTest($productIndexer, 'products', self::DEFAULT_PRODUCT_COUNT);
+    }
+
+    public function testProductAreSearchableIfIndexingIsDisabled()
+    {
+        setConfig('algoliasearch/credentials/enable_backend', '0');
+
+        $this->processQueryOneProduct();
```

Comment:
Why did you extract it to a method anyway, when you don't reuse it? Or do you plan to?